Examples/Projects/Arduino/BldcSimpleFOC/boards/zero.json

```diff
@@ -0,0 +1,59 @@
+{
+  "build": {
+    "arduino": {
+      "ldscript": "flash_with_bootloader.ld"
+    },
+    "core": "arduino",
+    "cpu": "cortex-m0plus",
+    "extra_flags": "-DARDUINO_SAMD_ZERO -D__SAMD21G18A__ -D LUOSHAL=ATSAMD21_ARDUINO",
```

Comment:
I think that you should not have this file and let Platformio get it for you.
Also you declared `LUOSHAL` twice (the other one is on your platformio.ini) You should remove this file.